streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -294,11 +306,13 @@ public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
         if (state == KafkaStreams.State.CREATED) {
-            for (final StreamThread thread : threads)
-                thread.start();
-
             setState(KafkaStreams.State.RUNNING);
 
+            for (final StreamThread thread : threads) {
+                threadState.put(thread.getId(), StreamThread.State.RUNNING);
```

Comment:
The initial state for the thread is `RUNNING` in the thread's constructor. I guess you're saying just query that, I'll do that, thanks.